cmdLine/src/main/scala/topgun/cmdline/AsmRecord.scala

```diff
@@ -0,0 +1,86 @@
+package topgun.cmdline
+
+import java.io.{IOException, InputStream, RandomAccessFile}
+import java.nio.file.{Files, Paths}
+import java.util.HashMap
+
+import org.objectweb.asm.ClassReader
+import org.objectweb.asm.tree.ClassNode
+
+import scala.collection.mutable
+
+
+object AsmRecord {
+  private var map: HashMap[String, ClassInfo] = _
+  private var buildDirectory: String = ""
+
+  def peek(): Unit = {
+    println(map)
+  }
+
+  private def getPackageName(path: String): String = {
+    var packageName: String =
+      path.substring(buildDirectory.length + 1).replace('/', '.')
+    packageName = packageName.substring(0, packageName.length - 6)
+    packageName
+  }
+
+  private def build(path: String): Unit = {
+    try {
+      val f: RandomAccessFile = new RandomAccessFile(path, "r")
+      val b: Array[Byte] = Array.ofDim[Byte](f.length.toInt)
+      f.readFully(b)
+      val reader: ClassReader = new ClassReader(b)
+      val classNode: ClassNode = new ClassNode()
+      reader.accept(classNode, 0)
+      val packageName: String = getPackageName(path)
+      val classInfo: ClassInfo = new ClassInfo(path, classNode.sourceFile)
+      map.put(packageName, classInfo)
+    } catch {
+      case _: IOException => {}
+    }
+  }
+
+  def constructRecord(buildDir: String): Unit = {
+    buildDirectory = buildDir
+    try {
+      val paths = Files.walk(Paths.get(buildDir))
+
+      paths.filter(path => Files.isRegularFile(path)).map(path => path.toString).filter(s => s.endsWith(".class"))
+        .forEach(path => AsmRecord.build(path))
+    } catch {
+      case e: IOException => e.printStackTrace()
+    }
+  }
+
+}
+
+class AsmRecord() {
+  //pass buildDirectory information as a constructor parameter
+  val map = mutable.HashMap.empty[String, ClassInfo]
+
+  // TODO: Make buildDirectory dynamic, by receiving values on object construction 
+  //set buildDirectory information with constructor parameter.
+  var buildDirectory =
+  "/home/battlefield/Documents/Projects/Java/HW/out/production"
+
+  AsmRecord.constructRecord(this.buildDirectory)
+
+  def getRecord(key: String, clazzLoader: ClassLoader): ClassInfo = map.getOrElseUpdate(key, getInfo(key, clazzLoader: ClassLoader))
+
+  def getInfo(clazzName: String, clazzLoader: ClassLoader): ClassInfo = {
+    val iStream: InputStream = clazzLoader.getResourceAsStream(clazzName.replace(".", "/") + ".class")
+    val reader: ClassReader = new ClassReader(iStream)
+    val classNode: ClassNode = new ClassNode()
+    reader.accept(classNode, 0)
+
+    //    val reader: ClassReader = new ClassReader(getClass.getClassLoader.getResourceAsStream(clazzName.replace('.', '/')))
+    //
+    //    val classNode: ClassNode = new ClassNode()
+    //    reader.accept(classNode, 0)
```

Comment:
If this code is dead it should be removed from the PR.

Reply:
removed it, it did the same thing as the block above.